daemon/algod/api/server/v2/handlers.go

```diff
@@ -784,6 +790,23 @@ func (v2 *Handlers) GetStatus(ctx echo.Context) error {
 		CatchpointAcquiredBlocks:    &stat.CatchpointCatchupAcquiredBlocks,
 	}
 
+	upgradePropose := string(stat.UpgradePropose)
+	votesToGo := uint64(stat.NextProtocolVoteBefore) - uint64(stat.LastRound)
+	if upgradePropose != "" && votesToGo > 0 {
```

Comment:
Turns out this is the wrong condition. This is closer but something still isn't right.
```suggestion
	votesToGo := uint64(stat.NextProtocolVoteBefore) - uint64(stat.LastRound)
	if stat.NextProtocolVoteBefore != 0 && votesToGo > 0 {
```

Reply:
Node-ui uses this condition:

> if m.Header.UpgradeState != (types.UpgradeState{})

Maybe that's the correct one?